pkg/parser/uri_test.go

```diff
@@ -0,0 +1,68 @@
+package parser
+
+import (
+	"testing"
+)
+
+type tests struct {
+	dsn    string
+	scheme string
+	path   string
+	query  string
+}
+
+var fTests = []tests{
+	{"", "", "", ""},
+	{"file:///some/path/to/file.csv?format=csv", "file", "/some/path/to/file.csv", "format=csv"},
+	{"file:///some/path/to/file.csv", "file", "/some/path/to/file.csv", ""},
+	{"os:stdout?format=csv", "os", "", "format=csv"},
+}
+
+var handledErrors = map[error]bool{
+	ErrEmptyDSN: true,
+}
+
+func TestParseURI(t *testing.T) {
+	for _, test := range fTests {
+		file, err := parseURI(test.dsn)
+		_ = file
+		// Check that we parse without any unhandled errors
+		if !handledErrors[err] && err != nil {
+			t.Error(
+				"Encountered error when parsing:",
+				err,
+			)
+		}
+		// Check that we throw an error for empty dsns
+		if err == ErrEmptyDSN {
```

Comment:
Can we move this into another test please there is way to much logic in this test func

Reply:
Done.